### PR TITLE
Shayan/feq 2336/redirect guest users to login page

### PR DESCRIPTION
**Pull request by shayan-deriv**

### CODE REVIEW

File: src/components/AppHeader/HeaderConfig.tsx [hunk 1]
```diff
@@ -81,14 +81,14 @@ export const MenuItems: MenuItemsConfig[] = [
...
         label: 'Reports',
     },
     {
-        as: 'button',
-        href: 'https://app.deriv.com/appstore/traders-hub',
+        as: 'a',
+        href: 'https://app.deriv.com/appstore/cashier',
         icon: <CashierLogo iconSize='xs' />,
         label: 'Cashier',
     },
```
Comment (shayan-deriv) [inline src/components/AppHeader/HeaderConfig.tsx:94]: this is not in this PR's scope but I just fixed it here

File: src/hooks/custom-hooks/useIsAdvertiser.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ const useIsAdvertiser = (): boolean => {
     const [isAdvertiser, setIsAdvertiser] = useState(!error && !isEmptyObject(data));
 
     useEffect(() => {
-        if (error && error?.error.code === ERROR_CODES.ADVERTISER_NOT_FOUND) {
+        if (error && error?.code === ERROR_CODES.ADVERTISER_NOT_FOUND) {
```
Comment (shayan-deriv) [inline src/hooks/custom-hooks/useIsAdvertiser.ts:15]: this is due to error message structure change in api-hooks library

File: src/pages/orders/screens/OrderDetails/OrderDetails.tsx [hunk 3]
```diff
@@ -80,7 +80,7 @@ const OrderDetails = () => {
     if (isLoading || (!orderInfo && !error)) return <Loader isFullScreen />;
 
     // TODO: replace with proper error screen once design is ready
-    if (error) return <Text>{error?.error?.message}</Text>;
+    if (error) return <Text>{error?.message}</Text>;
```
Comment (shayan-deriv) [inline src/pages/orders/screens/OrderDetails/OrderDetails.tsx:83]: this is due to error message structure change in api-hooks library

File: src/pages/orders/screens/OrderDetails/__tests__/OrderDetails.spec.tsx [hunk 4]
```diff
@@ -225,7 +225,7 @@ describe('<OrderDetails />', () => {
         (mockUseGet as jest.Mock).mockReturnValue({
             ...mockUseGet(),
             data: {},
-            error: { error: { message: 'error message' } },
+            error: { code: 'code', message: 'error message' },
```
Comment (shayan-deriv) [inline src/pages/orders/screens/OrderDetails/__tests__/OrderDetails.spec.tsx:228]: this is due to error message structure change in api-hooks library

File: src/providers/AdvertiserInfoStateProvider/AdvertiserInfoStateProvider.tsx [hunk 5]
```diff
@@ -1,8 +1,7 @@
 import { createContext, PropsWithChildren, useContext } from 'react';
-import { TSocketError } from 'types';
 
 type TContextValue = {
-    error: TSocketError<'p2p_advertiser_info'> | undefined;
+    error: { code: string; message: string } | undefined;
```
Comment (shayan-deriv) [inline src/providers/AdvertiserInfoStateProvider/AdvertiserInfoStateProvider.tsx:4]: this is due to error message structure change in api-hooks library

**Review by ali-hosseini-deriv — APPROVED** [review]

**Review by matin-deriv — APPROVED** [review]

**Review by farrah-deriv — APPROVED** [review]

**Pull request merged**